src/library/dao/trackdao.cpp

```diff
@@ -37,16 +36,14 @@ QMutex TrackDAO::m_sTracksMutex;
 
 enum { UndefinedRecordIndex = -2 };
 
-TrackCacheItem::TrackCacheItem(TrackPointer pTrack)
-        : m_pTrack(pTrack) {
-    DEBUG_ASSERT(m_pTrack);
+RecentTrackCacheItem::RecentTrackCacheItem(TrackPointer pTrack)
+        : m_pTrack(std::move(pTrack)) {
+    DEBUG_ASSERT(!m_pTrack.isNull());
 }
 
-TrackCacheItem::~TrackCacheItem() {
-    DEBUG_ASSERT_AND_HANDLE(!m_pTrack.isNull()) {
-        return;
-    }
-    // qDebug() << "~TrackCacheItem" << m_pTrack << "ID"
+RecentTrackCacheItem::~RecentTrackCacheItem() {
+    DEBUG_ASSERT(!m_pTrack.isNull());
```

Comment:
Why did you remove .._AND_HANDLE? Can't we remove the whole statement, since we check it in the constructor as well?